packages/ui/Drawing/Drawing.js

```diff
@@ -239,46 +256,55 @@ const DrawingBase = kind({
 
 			contextRef.current.closePath();
 			setIsDrawing(false);
+			currentLinesArray.push(currentLine);
+
+			currentLine = {
+				brushColor: '',
+				brushSize: '',
+				drawingTool: '',
+				ev: null,
+				fillColor: '',
+				points: []
+			};
+
+			actionsIndex++;
 		},
 
 		startDrawing: (event, {points, drawingTool}) => {
-			const {beginPointRef, contextRef, disabled, ev, setIsDrawing} = event;
+			const {beginPointRef, brushColor, brushSize, contextRef, disabled, ev, fillColor, setIsDrawing} = event;
 			const nativeEvent = ev.nativeEvent;
 			if (disabled) return;
-
+			if (lastAction === 'undo' || lastAction === 'redo') {
+				lastAction = 'draw';
+				currentLinesArray = [...currentLinesArray.slice(0, actionsIndex + 1)];
+			}
 			const {offsetX, offsetY} = nativeEvent;
 			contextRef.current.beginPath(); // start a canvas path
 			contextRef.current.moveTo(offsetX, offsetY); // move the starting point to initial position
 			points.push({x: offsetX, y: offsetY});
 
+			currentLine['points'].push({x: offsetX, y: offsetY});
+			currentLine['drawingTool'] = drawingTool;
+			currentLine['fillColor'] = fillColor;
+			currentLine['brushColor'] = brushColor;
+			currentLine['brushSize'] = brushSize;
+			currentLine['ev'] = ev;
```

Comment:
Could you please create a function that returns this so we don't repeat this code block?